app/main/routes.py

```diff
@@ -120,4 +120,11 @@ def check_file(excel_file: FileStorage) -> str | None:
 
 @bp.app_errorhandler(HTTPException)
 def http_exception(error):
-    return render_template(f"{error.code}.html"), error.code
+    error_templates = (401, 404, 429, 500, 503)
+
+    if error.code in error_templates:
+        return render_template(f"{error.code}.html"), error.code
+    else:
+        if Config.ENABLE_VALIDATION_LOGGING:
+            current_app.logger.info(f"Unhandled HTTP error {error.code} found.")
+        return render_template("500.html")
```

Comment:
We can also return a 500 status code with:

```
return render_template("500.html"), 500
```

Reply:
It's not actually a 500 error that has occurred in this instance, so this may be a bit misleading - maybe the name of the 500 template should be renamed to make it less confusing. This is returned for all other errors than the ones listed in the tuple above, but I can return error.status_code if we want the actual error code returned also?

Reply:
yep good idea - lets return the actual status code